apm-agent-plugins/apm-jms-plugin/src/main/java/co/elastic/apm/agent/jms/JmsInstrumentationHelperImpl.java

```diff
@@ -74,11 +96,16 @@ public Span startJmsSendSpan(Destination destination, Message message) {
 
         try {
             if (span.isSampled()) {
-                span.withName("JMS SEND to ");
-                addDestinationDetails(destination, span);
+                message.setStringProperty(JMS_TRACE_PARENT_PROPERTY, span.getTraceContext().getOutgoingTraceParentHeader().toString());
+                if (destinationName != null) {
+                    span.withName("JMS SEND to ");
+                    addDestinationDetails(null, destination, destinationName, span);
+                    if (isDestinationNameComputed) {
+                        message.setStringProperty(JMS_DESTINATION_NAME_PROPERTY, destinationName);
```

Comment:
why do we have to propagate that downstream and why only for temp queues?

Reply:
The JMS `Destination` APIs (`Queue` and `Topic`) provide a `getName()` method. When we create a tmp queue explicitly through the `createTmpQueue()` API on the sender side, its type will be indicative of that, and we know not to use `getName()` (as it will create high cardinality in transaction names and in the service map, once we have such). However, on the receiver side, the queue/topic type is not guaranteed to indicate that it is temp, as it doesn't necessarily know that it is a tmp queue, so it may use what it gets from `getName`, which will be incompatible with the sender span.